frontend/app/src/components/asset-manager/managed/ManagedAssetForm.vue

```diff
@@ -120,14 +120,18 @@ const clearFieldErrors = (fields: Array<keyof SupportedAsset>) => {
   fields.forEach(clearFieldError);
 };
 
-watch(address, async () => {
-  const sanitizedAddress = sanitizeAddress(get(address));
-  if (get(address) !== sanitizedAddress) {
+watch([address, evmChain], async ([addressVal, evmChain]) => {
+  if (!evmChain) {
+    return;
+  }
+
+  const sanitizedAddress = sanitizeAddress(addressVal);
+  if (addressVal !== sanitizedAddress) {
     set(address, sanitizedAddress);
```

Comment:
I didn't notice this before but can we make this logic here to a computed?
It is not a good approach for a watcher to change the watched.

Reply:
I remove this part and move the validation to the input directly.